File: wallet_syncronizer/pkg/database/wallet_token/wallet_token.go
package wallet_token

import (
	"time"
	token_db "wallet-syncronizer/pkg/database/token"
	"wallet-syncronizer/pkg/database/wallet"
)

func (WalletToken) TableName() string {
	return "WalletToken"
}

type WalletToken struct {
	WalletId       string         `json:"wallet_id" gorm:"column:WalletId;primaryKey;varchar(1024)"`
	Wallet         wallet.Wallet  `json:"-"`
	TokenId        string         `json:"token_id" gorm:"column:TokenId;primaryKey;varchar(1024);not null"`
	Token          token_db.Token `json:"-"`
	TokenAmount    float64        `json:"token_amount" gorm:"column:TokenAmount;not null"`
	TokenAmountHex string         `json:"token_amount_hex" gorm:"column:TokenAmountHex;varchar(1024);not null"`
	CreatedAt      time.Time      `json:"created_at" gorm:"column:CreatedAt;autoCreateTime"`
	UpdatedAt      time.Time      `json:"updated_at" gorm:"column:UpdatedAt;autoUpdateTime"`
}

File: wallet_syncronizer_node/database/wallet_token.js
const { DataTypes } = require('sequelize');
const Token = require('./token');
const Wallet = require('./wallet');

module.exports = (sequelize) => {
    const WalletToken = sequelize.define('WalletToken', {
        WalletId: {
            type: DataTypes.STRING,
            primaryKey: true,
        },
        TokenId: {
            type: DataTypes.STRING,
            primaryKey: true,
            allowNull: false,
        },
        TokenAmount: {
            type: DataTypes.FLOAT,
            allowNull: false,
        },
        TokenAmountHex: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        CreatedAt: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },
        UpdatedAt: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },
    }, {
        tableName: 'WalletToken',
        timestamps: false, // Puoi impostare questo su true se desideri timestamps
    });

    // Definisci le relazioni con i modelli Token e Wallet
    WalletToken.belongsTo(Token(sequelize), {
        foreignKey: 'TokenId',
        as: 'Token',
    });
    WalletToken.belongsTo(Wallet(sequelize), {
        foreignKey: 'WalletId',
        as: 'Wallet',
    });

    return WalletToken;
};
